drivers/SmartThings/matter-rvc/src/init.lua

```diff
@@ -28,213 +25,232 @@ if version.api < 10 then
+  local map = device:get_field(COMPONENT_TO_ENDPOINT_MAP) or {}
+  if map[component] then
+    return map[component]
+  else
+    return device.MATTER_DEFAULT_ENDPOINT
   end
-  device:set_field(field, mode_tags_of_supported_modes, { persist = true })
 end
 
-local function get_field_mode_tags_of_supported_mode(device, field_prefix, index)
-  local field = string.format("%s_mode_tags_%d", field_prefix, index)
-  return device:get_field(field)
+local function device_added(driver, device)
+  local run_mode_eps = device:get_endpoints(clusters.RvcRunMode.ID) or {}
+  local clean_mode_eps = device:get_endpoints(clusters.RvcCleanMode.ID) or {}
+  local component_to_endpoint_map = {
+    ["runMode"] = run_mode_eps[1],
+    ["cleanMode"] = clean_mode_eps[1]
+  }
+  device:set_field(COMPONENT_TO_ENDPOINT_MAP, component_to_endpoint_map, {persist = true} )
 end
 
-local function set_field_supported_modes(device, field_prefix, supported_modes)
-  local labels_field = string.format("%s_labels", field_prefix)
-  local labels_of_supported_modes = {}
-  device.log.info_with({hub_logs = true}, string.format("Supported modes: %s", utils.stringify_table(supported_modes)))
-  for i, mode in ipairs(supported_modes) do
-    if version.api < 10 then
-      clusters.RvcRunMode.types.ModeOptionStruct:augment_type(mode)
-    end
-    table.insert(labels_of_supported_modes, mode.elements.label.value)
-    set_field_mode_tags_of_supported_mode(device, field_prefix, i, mode.elements.mode_tags)
-  end
-  device:set_field(labels_field, labels_of_supported_modes, { persist = true })
+local function device_init(driver, device)
+  device:subscribe()
+  device:set_component_to_endpoint_fn(component_to_endpoint)
 end
 
-local function set_field_supported_modes_from_response(device, field_prefix, cluster_id, attribute_id, response)
-  for _, rb in ipairs(response.info_blocks) do
-    if rb.info_block.attribute_id == attribute_id and
-      rb.info_block.cluster_id == cluster_id and
-      rb.info_block.data.elements ~= nil then
-      set_field_supported_modes(device, field_prefix, rb.info_block.data.elements)
-      break
-    end
+local function do_configure(driver, device)
+  local clean_mode_eps = device:get_endpoints(clusters.RvcCleanMode.ID)
+  if #clean_mode_eps == 0 then
+    local profile_name = "rvc"
+    device.log.info_with({hub_logs=true}, string.format("Updating device profile to %s.", profile_name))
+    device:try_update_metadata({profile = profile_name})
   end
 end
 
-local function get_field_labels_of_supported_modes(device, field_prefix)
-  local field = string.format("%s_labels", field_prefix)
-  return device:get_field(field)
+local function info_changed(driver, device, event, args)
+  for cap_id, attributes in pairs(subscribed_attributes) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, attr in ipairs(attributes) do
+        device:add_subscribed_attribute(attr)
+      end
+    end
+  end
+  device:subscribe()
```

Comment:
```suggestion
if device.profile.id ~= args.old_st_store.profile.id then
    device:subscribe()
end
```
a little cleaner and does the same thing

Reply:
Thank you. I will modify it.